client/reader/stream/index.jsx

```diff
@@ -206,6 +207,7 @@ class ReaderStream extends Component {
 			showSelectedPost( {
 				...args,
 				postKey: postKey.isCombination ? keyForPost( args ) : postKey,
+				postObject: this.props.postsItems[ index ],
```

Comment:
Keying by `postKey` instead of `index` is a bit more resilient:
```js
this.props.postsItems[ postKey ]
```
Here `postsItems` would need to be an object instead of an array.

Reply:
The `postKey` can be reliably converted to a string with `keyToString` -- all the state selectors internally do this conversion.